### PR TITLE
fix(scdm/nemesis): enable RefreshMonkey for kube

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/VoFswDGm/2286-k8s-refreshmonkey

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Tested in here - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo/10/

File: sdcm/remote/base.py [hunk 1]
```diff
@@ -111,8 +111,15 @@ def sudo(self,
              new_session: bool = False,
              log_file: Optional[str] = None,
              retry: int = 1,
-             watchers: Optional[List[StreamWatcher]] = None) -> Result:
-        return self.run(cmd=cmd if self.user == "root" else f"sudo {cmd}",
+             watchers: Optional[List[StreamWatcher]] = None,
+             user: Optional[str] = 'root'
+             ) -> Result:
```
Comment (enaydanov) [inline sdcm/remote/base.py:116]: Looks pretty ugly for my taste.

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -632,22 +632,21 @@ def disrupt_nodetool_refresh(self, big_sstable=True, skip_download=False):
...
+            # Download the sstable files from S3
+            remote_get_file(node.remoter, sstable_url, sstable_file,
+                            hash_expected=sstable_md5, retries=2,
+                            user_agent=creds['user_agent'])
+            result = node.remoter.sudo("ls -t /var/lib/scylla/data/keyspace1/")
             upload_dir = result.stdout.split()[0]
-            node.remoter.run('sudo -u scylla tar xvfz {} -C /var/lib/scylla/data/keyspace1/{}/upload/'.format(
-                sstable_file, upload_dir))
+            node.remoter.sudo(
+                f'tar xvfz {sstable_file} -C /var/lib/scylla/data/keyspace1/{upload_dir}/upload/', user='scylla')
```
Comment (enaydanov) [inline sdcm/nemesis.py:644]: I think it's better to add `if node.is_docker():` condition here as for housekeeping mark and don't override `remoter.sudo()`  method it k8s remoter.

BTW, it's only two places in our codebase where we run `sudo -u`. Both times `sudo -u scylla`. Can you agree that it doen't worth of "remoter.sudo()" method complications?

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -632,22 +632,21 @@ def disrupt_nodetool_refresh(self, big_sstable=True, skip_download=False):
...
             # Scylla Enterprise 2019.1 doesn't support to load schema.cql and manifest.json, let's remove them
-            node.remoter.run(
-                'sudo rm -f /var/lib/scylla/data/keyspace1/{}/upload/schema.cql'.format(upload_dir))
-            node.remoter.run(
-                'sudo rm -f /var/lib/scylla/data/keyspace1/{}/upload/manifest.json'.format(upload_dir))
+            node.remoter.sudo(
+                'rm -f /var/lib/scylla/data/keyspace1/{}/upload/schema.cql'.format(upload_dir))
+            node.remoter.sudo(
+                'rm -f /var/lib/scylla/data/keyspace1/{}/upload/manifest.json'.format(upload_dir))
```
Comment (enaydanov) [inline sdcm/nemesis.py:649]: f-strings?

File: sdcm/nemesis.py [hunk 4]
```diff
@@ -2402,19 +2401,21 @@ def disrupt(self):
 class RefreshMonkey(Nemesis):
     disruptive = False
     run_with_gemini = False
+    kubernetes = True
 
     @log_time_elapsed_and_status
     def disrupt(self):
-        self.disrupt_nodetool_refresh()
+        self.disrupt_nodetool_refresh(big_sstable=False)
```
Comment (enaydanov) [inline sdcm/nemesis.py:2408]: But `big_sstable=True` by default. Something was wrong here?

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: sdcm/remote/base.py (hunk 1 above)
Comment (bentsi) [reply sdcm/remote/base.py:116]: agree

Comment (bentsi) [reply sdcm/remote/base.py:116]: `.... user: Optional[str] = 'root') -> Result:` is better style, let's use it

File: sdcm/remote/kubernetes_cmd_runner.py [hunk 5]
```diff
@@ -128,6 +130,33 @@ def _create_connection(self):
...
+             verbose: bool = True,
+             new_session: bool = False,
+             log_file: Optional[str] = None,
+             retry: int = 1,
+             watchers: Optional[List[StreamWatcher]] = None,
+             user: Optional[str] = 'root'
+             ) -> Result:
+        # Scylla is running as root under docker/kubernetes,
+        #   so we need this mapping to handle commands for scylla user
+        if user == 'scylla':
```
Comment (bentsi) [inline sdcm/remote/kubernetes_cmd_runner.py:146]: this shouldn't be done in the module, move this logic out to the relevant function that needs it

**Comment by dkropachev** [comment]
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo/27/

**Review by bentsi — APPROVED** [review]

**Pull request merged**